sky/provision/aws/instance.py

```diff
@@ -175,6 +187,24 @@ def cleanup_ports(
         'Values': [sg_name]
     }])
     if len(list(sgs)) != 1:
-        raise ValueError(f'Expected security group {sg_name} not found. '
-                         'Cannot cleanup ports.')
-    list(sgs)[0].delete()
+        logger.warning(f'Expected security group {sg_name} not found. '
+                       'Skip cleanup.')
+        return
+    backoff = common_utils.Backoff(initial_backoff=INITIAL_BACKOFF_SECONDS,
+                                   max_backoff_factor=MAX_BACKOFF_FACTOR)
+    for _ in range(MAX_ATTEMPTS):
+        try:
+            list(sgs)[0].delete()
+        except aws.botocore_exceptions().ClientError as e:
+            match = re.search(_DEPENDENCY_VIOLATION_PATTERN, str(e))
+            if match is None:
+                raise
+            if match.group(1) == 'DependencyViolation':
+                assert match.group(2) == 'DeleteSecurityGroup', match.group(2)
+                logger.debug(
+                    f'Security group {sg_name} is still in use. Retry.')
+                time.sleep(backoff.current_backoff())
+                continue
+        return
```

Comment:
Actually, why do we allow the case where the `match.group(1) != 'DependencyViolation'` to successfully return without raising an error or log? The deletion can fail in those cases as well right?
```suggestion
        except aws.botocore_exceptions().ClientError as e:
            match_str = 'An error occurred (DeleteSecurityGrou) when calling the DeleteSecurityGroup operation (.*): (.*)'
            if re.findall(match_str, str(e):
                logger.debug(
                    f'Security group {sg_name} is still in use. Retry.')
                time.sleep(backoff.current_backoff())
                continue
            raise
        return
```

Reply:
Oh I accidentally remove the raise when `match.group(1) != 'DependencyViolation'`... Thanks for point it out!